subsys/fs/shell.c

```diff
@@ -275,6 +275,80 @@ static int cmd_rm(const struct shell *sh, size_t argc, char **argv)
 	return err;
 }
 
+static int cmd_cp(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+	int close_err;
+	char path_src[MAX_PATH_LEN];
+	char path_dst[MAX_PATH_LEN];
+	struct fs_file_t file_src;
+	struct fs_file_t file_dst;
+	uint8_t buf[BUF_CNT];
+	ssize_t buf_len;
+	ssize_t num_written;
+
+	create_abs_path(argv[1], path_src, sizeof(path_src));
+	create_abs_path(argv[2], path_dst, sizeof(path_dst));
+
+	fs_file_t_init(&file_src);
+	fs_file_t_init(&file_dst);
+
+	err = fs_open(&file_src, path_src, FS_O_READ);
+	if (err) {
```

Comment:
IMO this should be boolean condition: `if (err != 0)`. Same for line 303.

Reply:
In my opinion, `if (err)` is better than `if (err != 0)`, for safety. Although I would personally use `if (0 != err)`.

Reply:
I do not think we have a standard for that, but 'err != 0' gives better clarity about type of variable being compared, specifically for people that are familiar with strongly typed languages that enforce type safety by requiring if expression to explicitly evaluate to boolean.

The '0 !=1 var' has some roots in secure programming, as an avoidance for mistakes where variable is assigned, rather than compared, due to a typo. It is unnatural to read and looks strange when variable has to be consecutively compared to different values.
It is equivalent of statement 'if constant is not a value of the variable', which sounds awkward.

Reply:
It's just unnatural for a while, it becomes normal in my experience. I've seen too many `if (val = 0)` in my life, I'm kind of scared of this mistake.

Reply:
According to rule 85 (https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html) the controlling expression of an if statement shall be a boolean expression.

Reply:
Got it. So is this fine or should I create another PR that fixes the entire file?

Reply:
This is fine. If you would like to create a PR that fixes the entire file that would be great.